### PR TITLE
Navbar Refactor

### PR DESCRIPTION
**Pull request by erunks**
## What
Huge refactor of the `<SiteNavbar />` component. Broke it down into smaller and more reusable pieces. Also updated the styling to be as close as possible to the wireframes.

## Changes
- Base ButtonLink: Created the `<ButtonLink />` component
- Core LoginButton: Created the `<LoginButton />` component.
- Core SignUpButton: Created the `<SignUpButton />` component.
- Core NavLinks: Created the `<NavLinks />` component.
- Core SearchBar: Created the `<SearchBar />` component.
- Components SiteNavbar: Refactored the `<SiteNavbar />` component after breaking it down into smaller pieces.
- Eslint: Added the `eol-last` rule.
- Project: Resolved all the linting errors.

### Steps for testing
N/A

### Images
| Desktop | Mobile |
| --- | --- |
| image | image |
| - | image |

### CODE REVIEW

**Review by maxcaplan — CHANGES_REQUESTED** [review]
This is looking great! Glad to finally have some proper organization. I just have a few requests for small structure tweaks and then we should be good to go.

File: components/core/LoginButton/LoginButton.vue
Comment (erunks) [reply components/core/LoginButton/LoginButton.vue:4]: Okay, I'll give this refactor a shot for sure. It definitely did feel clunky to write it like this, but that made the most sense at the time. I'll definitely give it another pass though.

File: components/base/ButtonLink.vue
Comment (erunks) [reply components/base/ButtonLink.vue:5]: We _can_, the thing that makes me hesitant about writing any of these paddings into the `class` is that it makes it hard to overwrite later on. Since I'm new to tailwind, I'm not sure if there's a good way to handle this or not. I might have to read more into tailwind though, cause this seems like it shouldn't be an issue.

File: components/base/ButtonLink.vue
Comment (erunks) [reply components/base/ButtonLink.vue:2]: I think it makes sense that not everything will need to be `<NuxtLink />`. At least for now, all the places that we're using this `<ButtonLink />` is a place where it should be a `<NuxtLink />`. I think what I'd like to see instead is a new `base` component (maybe just `<Button />`) that would expose those click events, but since there is currently no use for them, I'm not sure it makes sense to add it in this PR.

File: components/base/ButtonLink.vue
Comment (erunks) [reply components/base/ButtonLink.vue:5]: Okay, yeah. Reading more into the documentation, it looks like this definitely isn't an issue when done correctly. I'm for sure going to try and restructure these components even further to take advantage of the right way of doing things.
https://tailwindcss.com/docs/extracting-components

Comment (maxcaplan) [reply components/base/ButtonLink.vue:5]: That's a good point. I think that'll be a good way to go about things

File: components/base/ButtonLink.vue
Comment (maxcaplan) [reply components/base/ButtonLink.vue:2]: Yeah, makes sense to me. Leave that to a later PR

File: components/core/LoginButton/Login.vue
Comment (legrandgrand) [reply components/core/LoginButton/Login.vue:2]: As I said above probably CoreLoginButton doesn't sound correct since we will probably only use a LoginButton once, so probably need to change the file architecture fore those

**Review by legrandgrand — CHANGES_REQUESTED** [review]
Nice refactor! need to check the issues with eslint and the naming of CoreLoginButton and stuff, but it's overall very minor stuff anyway.

Out of curiosity, do we have husky running on the project to make sure prettier is applied  for every commits/something to make sure it is ran before every MR? I've seen request to have End of file line somewhere and prettier should check for that?

File: components/SiteNavbar.vue
Comment (erunks) [reply components/SiteNavbar.vue:108]: We do, it's the only reason I caught this. It's just not setup as a `pre-commit` hook or anything, but I think it should be

File: components/SiteNavbar.vue
Comment (erunks) [reply components/SiteNavbar.vue:100]: If we do want to disable one of them, I'd be in favour of disabling the `vue/order-in-components` as well, since I'd rather have all the keys ordered.

File: components/SiteNavbar.vue
Comment (erunks) [reply components/SiteNavbar.vue:65]: The only other place that I can think of us using the `<LoginButton />` is in the `AuthCards` which you can see in @maxcaplan's PR https://github.com/speedrun-website/speedrun.website/pull/14

File: components/core/LoginButton/Login.vue
Comment (erunks) [reply components/core/LoginButton/Login.vue:2]: I feel like there is a distinction between `base` and `core`, but I don't know how to explain it 😬 . We can just use one or the other though, if you think that makes more sense and is less confusing.

Comment (Matt-Hurd) [reply components/core/LoginButton/Login.vue:2]: Imo, core is more of a core component library that is imported, whereas base is more inheritance focused? When I hear base I think of "base class".

Comment (erunks) [reply components/core/LoginButton/Login.vue:2]: I think that's exactly what I was feeling, just didn't know how to articulate it. @legrandgrand what do you think about the distinction now?

Comment (legrandgrand) [reply components/core/LoginButton/Login.vue]: Sounds good to me then, we can use that if there is a difference

**Review by legrandgrand — APPROVED** [review]

**Review by maxcaplan — APPROVED** [review]
All my notes have been resolved. I'm good to have this merged once legrandgrand is happy with the changes!

**Review by Matt-Hurd — APPROVED** [review]

**Review by maxcaplan — CHANGES_REQUESTED** [review]
Oops, I just noticed that the spacing of the login signup and hamburger menu buttons gets messed up on mobile. It looks like you have them wrapped in an absolutely positioned div. I think the fix I would suggest would be change the layout to just use flex to position the buttons.

File: components/SiteNavbar.vue
Comment (maxcaplan) [inline components/SiteNavbar.vue:11]: You should add the `relative` class. Right now the absolute positioned buttons are positioned in relation to the max width of this element meaning it is offset to the right between the `md` and mobile breakpoints. Adding `relative` will position them with the containers margin.

File: components/SiteNavbar.vue
Comment (maxcaplan) [inline components/SiteNavbar.vue:62]: could you add the `items-center` class to this. Right now the child buttons height is being stretched to the height of this element making them taller than the other buttons. This will centre the buttons vertically and keep their heights consistent with the others.

**Review by maxcaplan — CHANGES_REQUESTED** [review]
Just two tiny changes, then you've got my seal of approval!

**Review by maxcaplan — APPROVED** [review]

**Pull request merged**
